datalad/metadata/aggregate.py

```diff
@@ -186,16 +193,130 @@ def _extract_metadata(agginto_ds, aggfrom_ds, db, to_save):
+            metasources['cn'] = {
+                'type': 'content',
+                'targetds': agginto_ds,
+                'dumper': json_py.dump2xzstream}
+
+    # check if we have the extracted metadata for this state already
+    # either in the source or in the destination dataset
+    metafound = {
+        s: d
+        # look in targetds last to not have to move things
+        # unnecessarily
+        for d in (aggfrom_ds, agginto_ds)
+        for s in metasources
+        # important to test for lexists() as we do not need to
+        # or want to `get()` metadata files for this test
+        # info on identity is sufficient
+        if op.lexists(
+            op.join(
+                d.path,
+                dirname(agginfo_relpath),
+                _get_obj_location(objid, s)))
+    } if not force_extraction else False
+
+    if not metafound:
+        lgr.info('FULLEXTRACT')
+        # no metadata found -> extract
+        # this places metadata dump files into the configured
+        # target dataset and lists them in `to_save`, as well
+        # as updates the `db` record for `aggfrom_ds`
+        return _extract_metadata(
+            agginto_ds,
+            aggfrom_ds,
+            db,
+            to_save,
+            objid,
+            metasources,
+            refcommit,
+            subds_relpaths)
+
+    # we did not actually run an extraction, so we need to
+    # assemble an aggregation record from the existing pieces
+    # that we found
+    # simple case: the target dataset has all the records already:
+    if all(d is agginto_ds for s, d in metafound.items()):
+        lgr.error('UNCHANGED')
+        # no change, use old record from the target dataset
+        db[aggfrom_ds.path] = old_agginfo
+        # no error
+        return False
+    else:
+        lgr.info('MVFORMSOURCE')
+        # we need to move the metadata dump(s) into the target dataset
+        objrelpaths = {
+            label: op.join(
+                dirname(agginfo_relpath),
+                _get_obj_location(objid, label))
+            for label in metafound
+        }
+        # make sure all the to-be-moved metadata records are present
+        # locally
+        aggfrom_ds.get(
+            # prep annotated path records to speed up the call
+            path=[dict(path=op.join(aggfrom_ds.path, p),
+                       parentds=aggfrom_ds.path,
+                       type='file')
+                  for p in objrelpaths],
+            result_renderer='disabled')
+
+        # actually copy dump files
+        for objrelpath in objrelpaths.values():
+            objpath = op.join(agginto_ds.path, objrelpath)
+            objdir = dirname(objpath)
+            if not exists(objdir):
+                makedirs(objdir)
+            # XXX TODO once we have a command that can copy/move files
+            # from one dataset to another including file availability
+            # info, this should be used here
```

Comment:
This would be lovely to have. And should not be that hard.

shutil.copy() + annex info reassignment + annex metadata duplication + done

or?